### DISCUSSION TITLE
How to ignore error when use refreshInterval?

### DISCUSSION THREAD

**Discussion by Sara2009** [vercel/swr#999] [Q&A]
I just what to handle the error when first to fetch data. If the fetching data is triggered by the refreshInterval. I don't want to do anything. How can I config the useSWR?

**Comment by ErnestBrandi** [comment]
Well you don't have to do anything then... Just don't handle the error as you said.

**Reply by Sara2009** [reply]
But it will also trigger onError function. I cannot figure out them.
```javascript
const { data, error } = useSWR(
    'api/fetchPKRankData',
    () => {
      return fetchData();
    },
    {
      refreshInterval: 15000,
      onError(err) {
          window.alert(err);
      }
    },
  );

````

**Comment by shuding** [comment]
It sounds weird to me to "ignore error", IMO every error should notify you. If you want to only show the error alert or notification **for the first time**, you should consider doing this (which is not in the scope of SWR):

```js
// global level
const errorOccurred = {}

// component
const { data, error } = useSWR(
    key,
    fetcher,
    {
      refreshInterval: 15000,
      onError(err) {
        if (errorOccurred[key]) return
        errorOccurred[key] = true
        window.alert(err)
      }
    },
  )
```

Hope that helps!

**Reply by Sara2009** [reply]
Thanks a lot. It works. I just want to show the error alert for the first time when open the page. The list data in the page needs to be updated quietly using refreshInterval. So i want to ignore the polling refresh error. But i find the refresh will stop if the fetcher returns a reject. But i need the refresh continues.

**Reply by shuding** [reply]
Then you can do this, instead of throwing you can wrap it with another layer to say it's data or error:

```js
const { data } = useSWR(
    key,
    async (url) => {
      try {
        return {
          data: await fetcher(url)
        }
      } catch (err) {
        return {
          error: err
        }
      }
    { refreshInterval: 15000 },
  )

const realData = data.data
const realError = data.error
```

Since `fetcher` will no longer throw, it will just keep polling. And you can handle data/error with `realData` and `realError`.

**Reply by Sara2009** [reply]
Thanks a lot. I'll try it.
